src/Linters/HHClientLintError.hack

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+enum ArcanistLintSeverityEnum: string as string {
+  ADVICE = 'advice';
+  AUTOFIX = 'autofix';
+  WARNING = 'warning';
+  ERROR = 'error';
+  DISABLED = 'disabled';
+}
+
+final class HHClientLintError extends LintError {
+
+  const type TJSONError = shape(
+    'descr' => string,
+    'severity' => ArcanistLintSeverityEnum,
+    'path' => string,
+    'line' => int,
+    'start' => int,
+    'end' => int,
+    'code' => int,
+    'bypass_changed_lines' => bool,
+    'original' => string,
+    'replacement' => string,
+  );
+
+  public function __construct(
+    HHClientLinter $linter,
+    private this::TJSONError $error,
+  ) {
+    parent::__construct($linter, $error['descr']);
+  }
+
+  <<__Override>>
+  public function getPosition(): (int, int) {
+    return tuple($this->error['line'], $this->error['end']);
+  }
+
+  <<__Override>>
+  public function getRange(): ((int, int), (int, int)) {
+    return tuple(
+      tuple($this->error['line'], $this->error['start']),
+      tuple($this->error['line'], $this->error['end']),
+    );
+  }
+
+  <<__Override>>
+  public function getBlameCode(): string {
+    return (string)$this->error['code'];
```

Comment:
This should be returning the hack source code that is responsible for the error, not the numeric error code.

It might be reasonable to change `LintError::getBlameCode`'s default implementation to use getRange() and read it `getFile()`, instead of requiring it to be manually implemented here.

That said, the numeric lint code does need to be exposed. HHAST's current linters are all one-linter-per-class, so it doesn't vary by lint error. This needs to change.

These needs to be reflected in:
- HHAST_IGNORE_ERROR[code] must work
- lint errors in the CLI
- lint errors in vscode

Searching for `get_class` and `getLinterName` is likely a good way to find both these places, and any others that are appropriate

Reply:
Do you mean `HH_IGNORE_ERROR[code]`? I am not feeling comfortable sharing the same marker format with hhast linters.

Reply:
Nope, `HHAST_IGNORE_ERROR[code]`; `HHAST_IGNORE_ERROR[LinterName]` is the existing suppression method used throughout.

Definitely don't want to re-use HH_ - they'd be removed when running `hh_client --remove-dead-fixmes`.

Reply:
But why? Suppose there is an open source library synced from internal repository, we would like the marker be understood by both internal CI and the hhast linter, right?

Reply:
In the past, that's not been a major concern due to there being no shared linters; with this, it's probably worth revisiting #351, but whatever's done should be consistent for all linters available via hhast - people should be able to use the same syntax regardless of if a linter is implemented in Hack or OCaml

Reply:
Fair enough! I agree https://github.com/hhvm/hhast/pull/351 should be a good direction. But I would not address other marker format in this PR, instead `HHAST_*` markers should be a good start in this PR.

Reply:
See https://github.com/hhvm/hhast/pull/388 for `HHAST_IGNORE_ERROR[code]`

Reply:
> lint errors in the CLI
> lint errors in vscode

Will be implemented in next couple of PRs